vumi-entrypoint.sh

```diff
@@ -1,39 +1,62 @@
 #!/usr/bin/env sh
 set -e
 
-TWISTD_COMMAND="${TWISTD_COMMAND:-vumi_worker}"
-
-WORKER_CLASS_OPT="${WORKER_CLASS:+--worker-class $WORKER_CLASS}"
-CONFIG_OPT="${CONFIG_FILE:+--config $CONFIG_FILE}"
-
-AMQP_OPTS=""
-if [ -n "$AMQP_HOST" ]; then
-  AMQP_OPTS="--hostname $AMQP_HOST \
-    --port ${AMQP_PORT:-5672} \
-    --vhost ${AMQP_VHOST:-/} \
-    --username ${AMQP_USERNAME:-guest} \
-    --password ${AMQP_PASSWORD:-guest}"
-fi
-
-SENTRY_OPT="${SENTRY_DSN:+--sentry $SENTRY_DSN}"
-
-SET_OPTS=$(env \
-  | grep ^VUMI_OPT_ \
-  | sed -e 's/^VUMI_OPT_//' -e 's/=/ /' \
-  | awk '{printf("%s=%s:%s ", "--set-option", tolower($1), $2);}')
+# If `python` isn't available, we should use `pypy`
+python=$(command -v python > /dev/null && echo python || echo pypy)
 
 # Generate Twisted's plugin cache just before running -- all plugins should be
 # installed at this point. Twisted is installed site-wide, so the root user is
 # needed to perform this operation. See:
 # http://twistedmatrix.com/documents/current/core/howto/plugin.html#plugin-caching
-python -c 'from twisted.plugin import IPlugin, getPlugins; list(getPlugins(IPlugin))'
-
-exec su-exec vumi \
-  twistd --nodaemon --pidfile="" \
-    $TWISTD_COMMAND \
-    $WORKER_CLASS_OPT \
-    $CONFIG_OPT \
-    $AMQP_OPTS \
-    $SENTRY_OPT \
-    $SET_OPTS \
-    "$@"
+"$python" -c 'from twisted.plugin import IPlugin, getPlugins; list(getPlugins(IPlugin))'
+
+is_twistd_command() {
+	local cmd="$1"; shift
+	"$python" - <<EOF
+import sys
+from twisted.plugin import getPlugins
+from twisted.application.service import IServiceMaker
+sys.exit(0 if '$cmd' in [p.tapname for p in getPlugins(IServiceMaker)] else 1)
```

Comment:
Twisted is complicated... but I found this here: https://github.com/twisted/twisted/blob/twisted-17.5.0/src/twisted/application/twist/_options.py#L174-L182